### PR TITLE
Enable KF export

### PR DESCRIPTION
**Pull request by HENDRIX-ZT2**
@niftools/blender-niftools-addon-reviewer 

# Overview
Improve animation export

##  Detailed Description
**[List of functional updates]**
- Moves KF export to own operator
- makes xnif / xkf version dependant for MW,
- removes copypasta for skyrim kf export.
- convert more format() strings to fstrings
- move scale toaster correction to nif common

## Fixes Known Issues
**[Ordered list of issues fixed by this PR]**

## Documentation
**[Overview of updates to documentation]**

## Testing
**[Overview of testing required to ensure functionality is correctly implemented]**

### Manual
**[Set of steps to manually verify updates are working correctly]**

### Automated
**[List of tests run, updated or added to avoid future regressions]**

## Additional Information
**[Anything else you deem relevant]**

### CODE REVIEW

File: io_scene_niftools/kf_export.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+        NifLog.info(f"Exporting {NifOp.props.filepath}")
+
+        # extract directory, base name, extension
+        directory = os.path.dirname(NifOp.props.filepath)
+        filebase, fileext = os.path.splitext(os.path.basename(NifOp.props.filepath))
+
+        prefix = "x" if bpy.context.scene.niftools_scene.game in ('MORROWIND',) else ""
+        self.version, data = scene.get_version_data()
+        NifData.init(data)
```
Comment (neomonkeus) [inline io_scene_niftools/kf_export.py:74]: Store in KFData

File: io_scene_niftools/kf_export.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        ext = ".kf"
+        NifLog.info(f"Writing {prefix}{ext} file")
+
+        data.roots = [kf_root]
+        data.neosteam = (bpy.context.scene.niftools_scene.game == 'NEOSTEAM')
+
+        # scale correction for the skeleton
+        if bpy.context.scene.niftools_scene.game in ('SKYRIM'):
+            toaster = pyffi.spells.nif.NifToaster()
+            toaster.scale = round(1 / NifOp.props.scale_correction)
```
Comment (neomonkeus) [inline io_scene_niftools/kf_export.py:94]: pull out `1 / NifOp.props.scale_correction` to local variable `scale`
as also used on line 96

File: io_scene_niftools/modules/nif_export/animation/transform.py [hunk 3]
```diff
@@ -51,343 +51,287 @@
 
 class TransformAnimation(Animation):
 
-    def __init__(self):
```
Comment (neomonkeus) [inline io_scene_niftools/modules/nif_export/animation/transform.py:54]: Looks like spaces are off

File: io_scene_niftools/kf_export.py (hunk 1 above)
Comment (HENDRIX-ZT2) [reply io_scene_niftools/kf_export.py:74]: I would, but this is at times also used to access data version, during the transform functions which are also used by nif export + anim. Very ugly.

File: io_scene_niftools/kf_export.py (hunk 2 above)
Comment (HENDRIX-ZT2) [reply io_scene_niftools/kf_export.py:94]: Should probably be moved out to a function anyway so nif and kf export use the same

File: io_scene_niftools/modules/nif_export/animation/transform.py (hunk 3 above)
Comment (HENDRIX-ZT2) [reply io_scene_niftools/modules/nif_export/animation/transform.py]: Changed back to spaces.

File: io_scene_niftools/modules/nif_export/animation/transform.py [hunk 4]
```diff
@@ -154,50 +124,14 @@ def export_kf_root(self, b_armature=None):
             kf_root.text_keys = anim_textextra
             kf_root.cycle_type = NifFormat.CycleType.CYCLE_CLAMP
             kf_root.frequency = 1.0
-            kf_root.start_time = bpy.context.scene.frame_start * bpy.context.scene.render.fps
-            kf_root.stop_time = (bpy.context.scene.frame_end - bpy.context.scene.frame_start) * bpy.context.scene.render.fps
+            kf_root.start_time = scene.frame_start * scene.render.fps
```
Comment (neomonkeus) [inline io_scene_niftools/modules/nif_export/animation/transform.py:127]: 👍

**Review by neomonkeus — APPROVED** [review]

File: io_scene_niftools/kf_export.py (hunk 1 above)
Comment (neomonkeus) [reply io_scene_niftools/kf_export.py:74]: Can revisit at a later point

File: io_scene_niftools/kf_export.py (hunk 2 above)
Comment (neomonkeus) [reply io_scene_niftools/kf_export.py:94]: Would be a good improvement alright

Comment (HENDRIX-ZT2) [reply io_scene_niftools/kf_export.py]: Done!

File: io_scene_niftools/kf_export.py (hunk 1 above)
Comment (HENDRIX-ZT2) [reply io_scene_niftools/kf_export.py:74]: Yeah I've added a todo. This definitely needs to be addressed.

File: io_scene_niftools/nif_export.py [hunk 5]
```diff
@@ -288,14 +243,10 @@ def execute(self):
             """
 
             # apply scale
+            data.roots = [root_block]
             scale_correction = bpy.context.scene.niftools_scene.scale_correction
             if abs(scale_correction) > NifOp.props.epsilon:
```
Comment (neomonkeus) [inline io_scene_niftools/nif_export.py:248]: 💯

Comment (HENDRIX-ZT2) [reply io_scene_niftools/nif_export.py:248]: Not sure when to use bpy.context.scene.niftools_scene.scale_correction and NifOp.props.scale_correction lol

**Review by neomonkeus — APPROVED** [review]

File: io_scene_niftools/kf_export.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+class KfExport(NifCommon):
+
+    def __init__(self, operator, context):
+        NifCommon.__init__(self, operator, context)
+
+        # Helper systems
+        self.transform_anim = TransformAnimation()
+
+    def execute(self):
+        """Main import function."""
```
Comment (Candoran2) [inline io_scene_niftools/kf_export.py:64]:
```suggestion
        """Main export function."""
```

**Review by neomonkeus — APPROVED** [review]

**Review by Candoran2 — APPROVED** [review]

**Review by neomonkeus — APPROVED** [review]

**Pull request merged**